src/clj/doit/todo/handler.clj

```diff
@@ -28,18 +28,18 @@
 
 (defn update-todo [request]
   (let [body (parse-body (:body request))
-        parsed-body (s/conform ::spec/update-params body)
         id (Integer. (get-in request [:route-params :id]))
-        todo (todo-db/retrieve-todo id)
-        updated-todo (select-keys (merge todo parsed-body) [:content :id :done])]
+        todo (todo-db/retrieve-todo id)]
     (if-not todo
       (wrap-response {:error (format "todo with id %s not found" id)} 404)
-      (if (= parsed-body ::s/invalid)
-        (wrap-response {:error (s/explain-str ::spec/update-params body)} 400)
-        (-> updated-todo
-            todo-db/update-todo!
-            (select-keys [:content :id :done])
-            (wrap-response 200))))))
+      (let [parsed-body (s/conform ::spec/update-params body)]
+        (if (= parsed-body ::s/invalid)
+                (wrap-response {:error (s/explain-str ::spec/update-params body)} 400)
+                (let [updated-todo (select-keys (merge todo parsed-body) [:content :id :done])]
+                  (-> updated-todo
+                      todo-db/update-todo!
+                      (select-keys [:content :id :done])
+                      (wrap-response 200))))))))
```

Comment:
Extracting out a `update-todo*` with non-ring, non-controller specific code might make this code easier to read.